api/gooseai.py

```diff
@@ -45,6 +48,9 @@ def get_engine(self) -> GooseAIEngines:
                     return engine
             except Exception as e:
                 log.error(self.class_name, _msg=f"Got error checking if {engine.name} is online.", e=e)
+                loop = asyncio.get_running_loop()
+                loop.create_task(utils.log_error(f"Got error checking if {engine.name} is online."))
+                loop.create_task(utils.log_exception(e))
```

Comment:
Some thoughts:

* Looks like this class should not know about logging details, including asyncio.
* A new error logging service "errorLogger" might be warranted, and then injected in the constructor
* I wonder what the difference between `log.error` and `utils.log_error` is. Probably that is another logging detail that should not be visible in this class, such as the usage of multiple logging channels.
* utils.... having a dedicated logger or error logger will fix Interface Segregation and Single Responsibility Priniple issues
* Failing all of that you can still likely extract a private method: `logOnlineCheckFailed` or some such

Reply:
`log.error` goes to the actual server logs on the host.  `utils.log_error` sends it to the discord error log channel to notify devs immediately of an important error impacting production.

Reply:
Yeah ok. So indeed different logging channels and other logging details. That should not be visible in the GooseAI code. Instead the GooseAI code should just be exposed to a logger interface, with the application itself taking care of building a correctly configured logger instance and passing it to GooseAI.

Not sure how helpful my explanation is. Might be too high level if you have never done something like that.

Reply:
I understand what you're saying, I just don't see the benefit in refactoring the entire codebase again to change logging.  @chriscanal thoughts?

Reply:
I was not suggesting you refactor the whole codebase. Much of the issues I wrote about already existed, and I'm sure there are many more instances of them that have nothing to do with logging.

A less drastic change is to make the things you are already touching better. Or at least not make them worse. As it currently is, the changes move the code further away from good design. Personally, I would take this as an opportunity to introduce the missing interface and use it in GooseAI. Then if someone feels like it, they can refactor the other code.

Reply:
I would personally prefer if @tayler6000 merged this PR for now and @JeroenDeDauw created a new PR with the improvements you have in mind so we can understand more precisely what exactly you propose to change at first. What do you think?

Reply:
Yeah I think that's the right call. First get the bot working again, then improvements

Reply:
My preference is also to demonstrate such changes rather than provide theoretical explanation. This is why I created https://github.com/robertskmiles/stampy/issues/174 as a response to this conversation.